R/plotutils.R

```diff
@@ -159,7 +157,7 @@ errorpos <- function(dx,errsum.method="linear") {
 #' plotwitherror(x, y, dy)
 #' 
 #' @export plotwitherror
-plotwitherror <- function(x, y, dy, ylim, dx, xlim, mdx, mdy, errsum.method="linear.quadrature", rep=FALSE, col="black", ...) {
+plotwitherror <- function(x, y, dy, ylim = NULL, dx, xlim = NULL, mdx, mdy, errsum.method="linear.quadrature", rep=FALSE, col="black", ...) {
```

Comment:
What is the point of this? Isn't `missing` just fine?

Reply:
Basically this is a convenience for client code. Imagine you have branching in an analysis code which computes something for `ylim` if some condition(s) is(are) met or delegates its calculation to the algorithm within `plotwitherror` if the condition(s) is(are) not met.

Using `missing()`, the branching needs to produce two different calls of `plotwitherror`. On the other hand, when `NULL` is used instead, you have a single call and just modify whether you pass `NULL` or not.

Reply:
This reminds me to review the `plotwitherror` documentation one of these days because it's not fully representative of the current state of the function.

Reply:
Of course, one could achieve something similar using `do.call` and having the branching produce different argument lists.

Reply:
Indeed, using `NULL` makes this more convenient for the caller.